src/cssls.ts

```diff
@@ -7,35 +7,41 @@ import {
 
 import sortCollectionSet from './util/sortCollectionSet';
 
+type Cssls = (
+  XtX: Matrix,
+  XtY: Matrix,
+  Pset: number[][] | null,
+  l: number,
+  p: number,
+) => Matrix;
```

Comment:
better define the types directly on the function declaration instead of this indirection.